app/models/profile.rb

```diff
@@ -14,7 +14,7 @@ class Profile < ApplicationRecord
 
   has_many :profile_rules, dependent: :delete_all
   has_many :rules, through: :profile_rules, source: :rule
-  has_many :profile_hosts, dependent: :delete_all
+  has_many :profile_hosts, dependent: :destroy
```

Comment:
I'm pretty sure this has performance implications. We originally had destroy here and changed it to delete_all. I can't quite remember why though.

Reply:
https://github.com/RedHatInsights/compliance-backend/pull/99 no apparent reason, this only runs when the profile or report are deleted